src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -614,18 +615,13 @@ private AuthenticationResult processTokenResponse(HttpWebResponse webResponse) {
 
         // Set correlationId in the result
         if (correlationIdInHeader != null && !correlationIdInHeader.isEmpty()) {
-            try {
-                UUID correlation = UUID.fromString(correlationIdInHeader);
-                if (!correlation.equals(mRequest.getCorrelationId())) {
-                    Logger.w(TAG, "CorrelationId is not matching", "",
-                            ADALError.CORRELATION_ID_NOT_MATCHING_REQUEST_RESPONSE);
-                }
-
-                Logger.v(TAG, "Response correlationId:" + correlationIdInHeader);
-            } catch (Exception ex) {
-                Logger.e(TAG, "Wrong format of the correlation ID:" + correlationIdInHeader, "",
-                        ADALError.CORRELATION_ID_FORMAT, ex);
+            UUID correlation = UUID.fromString(correlationIdInHeader);
```

Comment:
This shouldn't be fatal.  In fact, it should only be logged as a warning.  We should reverse the logic.

1. mRequest.getCorrelationId() should be converted to a string.
2. That string should be compared to the string correlation id returned from the server.
3. If they are not equal then we can log a warning that contains the correlation id returned from the server.

The correlation id is pure diagnostics and the fact that something unexpected might happen with the correlation id should not affect the success or failure of a request in any way.  Changing the logic to the above will ensure that no exceptions are thrown unless it is truly an exceptional case.  If it were me, I would also move this code out to a separate method named something like 'warnOnCorrelationIdMismatch', but I'm not particularly passionate about it.